### PR TITLE
feat: introduced @rematch/typed-state plugin

### PR DESCRIPTION
**Pull request by semoal**
I think it's a cool addition to javascript projects that want to get some safe access to his stores.

### CODE REVIEW

**Comment by semoal** [comment]
@tianzhich we you have time there's no hurry if you can take a look to typings, i think if we introduce together this plugin with select plugin won't work because typescript doesn't know merge multiple declare module '@rematch/core'

File: packages/typed-state/src/index.ts [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+				)
+			}
+			return called
+		},
+	}
+}
+
+export default typedStatePlugin
+
+declare module '@rematch/core' {
```
Comment (tianzhich) [inline packages/typed-state/src/index.ts:99]: Do you mean this type declaration won't work?

Comment (tianzhich) [reply packages/typed-state/src/index.ts:99]: From the [TS docs](https://www.typescriptlang.org/docs/handbook/declaration-merging.html#introduction):
> Any number of declarations can be merged; it’s not limited to just two declarations

Comment (semoal) [reply packages/typed-state/src/index.ts:99]: This type declaration won't work. Loses the selectors type, like typescript isn't deep merging.
Screenshot 2021-03-02 at 09 48 31

**Pull request merged**